### PR TITLE
feat(step_4): add statuses

### PR DESCRIPTION
**Pull request by AlexMomot-717**

### CODE REVIEW

File: task_manager/statuses/views.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import get_object_or_404, redirect, render
+from django.utils.translation import gettext as _
+from django.views import View
+from task_manager.statuses.forms import StatusForm
+from task_manager.statuses.models import Status
+from task_manager.users.views import UserLoginRequiredMixin
+
+
+class StatusesListView(View):
```
Comment (svntmr) [inline task_manager/statuses/views.py]: I guess we have to add `UserLoginRequiredMixin` here as task description states:

> Возможность просматривать, создавать, обновлять и удалять статусы доступна только залогиненным пользователям

File: task_manager/statuses/views.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        return render(
+            request,
+            "statuses_list.html",
+            context={
+                "statuses": statuses,
+            },
+        )
+
+
+class StatusFormCreateView(View):
```
Comment (svntmr) [inline task_manager/statuses/views.py]: I guess we have to add `UserLoginRequiredMixin` here as task description states:

> Возможность просматривать, создавать, обновлять и удалять статусы доступна только залогиненным пользователям

File: task_manager/templates/base.html [hunk 3]
```diff
@@ -28,13 +28,15 @@
...
           <ul class="navbar-nav mr-auto">
             <li class="nav-item">
-              <a class="nav-link" href="">{% translate "Statuses" %}</a>
+              <a class="nav-link" href="/statuses/"
+                >{% translate "Statuses" %}</a
+              >
             </li>
             <li class="nav-item">
-              <a class="nav-link" href="">{% translate "Labels" %}</a>
+              <a class="nav-link" href="/labels/">{% translate "Labels" %}</a>
```
Comment (svntmr) [inline task_manager/templates/base.html]: wdyt, does it make sense not to add `href` to this link as this functionality is not completed yet?

File: task_manager/templates/base.html [hunk 4]
```diff
@@ -28,13 +28,15 @@
...
             </li>
             <li class="nav-item">
-              <a class="nav-link" href="">{% translate "Tasks" %}</a>
+              <a class="nav-link" href="/tasks/">{% translate "Tasks" %}</a>
```
Comment (svntmr) [inline task_manager/templates/base.html]: wdyt, does it make sense not to add `href` to this link as this functionality is not completed yet?

File: task_manager/tests/test_status.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    route = f"/statuses/{status_id}/edit/"
+
+    # when
+    response = client.get(route)
+
+    # then
+    assert response.status_code == 404
+
+
+@freeze_time("15.07.2024 17:00")
```
Comment (svntmr) [inline task_manager/tests/test_status.py]: wdyt, does it make sense to create a [session-scope](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#fixture-scopes) [autouse fixture](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#autouse-fixtures-fixtures-you-don-t-have-to-request) to set the time for all the tests?

File: task_manager/tests/conftest.py [hunk 6]
```diff
@@ -1,7 +1,15 @@
...
 from django.test import Client
+from task_manager.users.models import ServiceUser
 
 
 @pytest.fixture
 def client() -> Client:
     return Client()
+
+
+@pytest.fixture
```
Comment (svntmr) [inline task_manager/tests/conftest.py:14]: great idea!

**Review by svntmr — CHANGES_REQUESTED** [review]

File: task_manager/tests/conftest.py [hunk 7]
```diff
@@ -1,7 +1,15 @@
...
 from django.test import Client
+from task_manager.users.models import ServiceUser
 
 
 @pytest.fixture
 def client() -> Client:
     return Client()
+
+
+@pytest.fixture()
```
Comment (svntmr) [inline task_manager/tests/conftest.py]: [optional] not sure if you need to call it if you don't intend to provide any extra parameters to the fixture

**Review by svntmr — APPROVED** [review]

File: task_manager/tests/conftest.py [hunk 8]
```diff
@@ -1,7 +1,24 @@
...
+
+
+@pytest.fixture
+def service_user() -> ServiceUser:
+    return ServiceUser.objects.create(
+        first_name="ilon", last_name="mask", username="ilon-mask", password="%%%%%"
+    )
+
+
+@pytest.fixture(autouse=True)
```
Comment (svntmr) [inline task_manager/tests/conftest.py]: in the current implementation it's going to be called for each test, wdyt does it make sense to call it once for the test execution session using [session-scope](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#fixture-scopes) fixture?

**Review by svntmr — APPROVED** [review]

File: task_manager/tests/conftest.py (hunk 8 above)
Comment (AlexMomot-717) [reply task_manager/tests/conftest.py]: done

**Pull request merged**